ADAPTSM-89 | @Lyafel | Membership card setup

Lyafel opened the pull request:
# NOT READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- TL;DR - what's this PR for?

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to...
3. Verify...

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

Lyafel commented:
@rebeccahongsf I added the card on /membership/saacard but it's a wip for now, and I have some questions.

- I was imagining this would be a nextjs page but looks like the header and footer should be comming from a storyblok page so I'm assuming now that this should be a regular component instead and added through storyblok?
- What are the urls for the links on the right
- The ticket only has the gsb images, still need alum and affiliate logo and bg
- What should I default to for the example card be. Should I reuse the static image used currently with the word Example on it? If so could you please share it with me?

Lyafel commented on src/utilities/mocks/fullgg.js:
```diff
@@ -272,4 +272,8 @@ export const fullggMockData = {
     access_method: 'GET',
   },
   affiliations: affiliationsMockData,
+  membership: {
```
@rebeccahongsf  Is this the correct way to add data to the mock user?  The reason I'm asking is that I can see this rendered on my local but not on the build, so I wonder if there's a step of the process I'm missing

rebeccahongsf replied:
That's a great question, @Lyafel! From my understanding. we currently do not have that environment variable(`MEGAPROFILE_MOCK = 'true'`) in the Netlify builds. Let me follow up with the team and see if there's a way we can turn it on for our dev previews.

rebeccahongsf replied:
And I'm back! We can add the environment variable(`MEGAPROFILE_MOCK = 'true'`) to the `netlify.toml` so we can test the mockserver data in this PR (we'll revert the `netlify.toml` before merging).

rebeccahongsf commented:
Thank you for working on this @Lyafel ! The client might request some adjustments for the card design (i.e. font color from white to black depending on accessibility, etc.). I'll make sure to update the ticket once if and when that occurs. 

In response to your awesome questions:
> * I was imagining this would be a nextjs page but looks like the header and footer should be comming from a storyblok page so I'm assuming now that this should be a regular component instead and added through storyblok?

Your assumptions are correct! Let's move this from a page into a component that can be added to a Basic Page in Storyblok for now. 

> * What are the urls for the links on the right

The client is still finalizing the URL copy and links. When building out the Digital Membership Card Component in StoryBlok, would you be able to add a block field limited to `ctaGroup` component so that the Content authors can manually add the links?

> * The ticket only has the gsb images, still need alum and affiliate logo and bg

Ack! My apologies on the delay for those images — I'll follow up in the ticket.

> * What should I default to for the example card be. Should I reuse the static image used currently with the word Example on it? If so could you please share it with me?

The default should be the example card from the screenshot — I'll follow up in the ticket with the original static image as well.

Lyafel commented:
@rebeccahongsf Moved the membership card to storyblok https://deploy-preview-589--stanford-alumni.netlify.app/editor/?access_key=EqUgXbgr6fMOWrXmyugXiwtt&path=membership/membership-card (the edit link works, figuring out why does the regular netlify link error out). 

Looks like ctaGroup is working but the icons it accepts don't include the cash icon for the last link on the design, I could only find it in HeroIcon. 
Also it seems that with moving the links to use ctaGroup, the links on the logged out page will still remain static? Could you please share the urls, I assume they are the ones used on alumni now in the provided site, but I still am unable to login.

rebeccahongsf commented:
> @rebeccahongsf Moved the membership card to storyblok
> 
> Looks like ctaGroup is working but the icons it accepts don't include the cash icon for the last link on the design, I could only find it in HeroIcon. Also it seems that with moving the links to use ctaGroup, the links on the logged out page will still remain static? Could you please share the urls, I assume they are the ones used on alumni now in the provided site, but I still am unable to login.

Hello @Lyafel ! Apologies for taking a while to get back to you (I was out sick the past two days) and the lack of clarity around the links.

Thank you so much for the work you've done on this! 

Let's put the URLs down as `https://alumni.stanford.edu/membership` for now since 
- the `View membership benefits` are outlined at that public facing URL
- I'll be handling the `Print your membership card` link in [another ticket](https://stanfordits.atlassian.net/browse/ADAPTSM-91)
- the `Make an installment payment online` page has not been built out yet (we're awaiting the client response). 

I'll follow up with design about the icon selections. Once that has been updated, I can review the PR and we can handle the additional link adjustments in another PR. Please let me know if additional clarity is needed. Thanks again! 😃

rebeccahongsf commented:
And I'm circling back now @Lyafel — I've created the Membership Card component as a Protected Content Item ([see on SB editor here](https://app.storyblok.com/#/me/spaces/184821/stories/0/0/270688040?env=TEMP:PR%23589)) and updated the [Membership Card page](https://app.storyblok.com/#/me/spaces/184821/stories/0/0/268210327/blok/f53342b9-4f9a-40ab-a8c9-d12ca236a6fc?env=TEMP:PR%23589). I believe that should resolve the authentication issue around the links. I also added the font awesome version of the cash icon (`money-bill`) while I was editing in SB. 😃

I logged in as `zguan` to review the page and it looks good so far: https://deploy-preview-589--stanford-alumni.netlify.app/membership/membership-card/
 
Let me know your thoughts and if this PR is ready for review!

Lyafel commented:
> And I'm circling back now @Lyafel — I've created the Membership Card component as a Protected Content Item ([see on SB editor here](https://app.storyblok.com/#/me/spaces/184821/stories/0/0/270688040?env=TEMP:PR%23589)) and updated the [Membership Card page](https://app.storyblok.com/#/me/spaces/184821/stories/0/0/268210327/blok/f53342b9-4f9a-40ab-a8c9-d12ca236a6fc?env=TEMP:PR%23589). I believe that should resolve the authentication issue around the links. I also added the font awesome version of the cash icon (`money-bill`) while I was editing in SB. 😃
> 
> I logged in as `zguan` to review the page and it looks good so far: https://deploy-preview-589--stanford-alumni.netlify.app/membership/membership-card/
> 
> Let me know your thoughts and if this PR is ready for review!

Thanks! I changed the grid width, should look a bit better now.  
I guess I assumed that the static example card will be shown to the non logged in users as well.
Aside from the missing assets for cards other that gsb it should be good for a first review

rebeccahongsf reviewed (changes requested):
Hello @Lyafel ! Thank you for your patience. Everything looks good so far! 

Just a few style updates — Bhavika recently updated the [Figma mocks](https://www.figma.com/file/BfEeWa39C6xT6muvdkBXsw/Membership-2022%2F23?node-id=2556%3A177745&t=bb97hxmb6tpVxqAa-1) to include responsive design. Would you be able to implement these changes?
![image](https://user-images.githubusercontent.com/34019925/223829950-5185e1c6-8702-4c86-b6f1-aa4381324f79.png)


Please do not hesitate to let me know if you have any questions. Thank you! 😃
